src/main/java/org/c02e/jpgpj/Ring.java

```diff
@@ -311,9 +311,13 @@ else if (packet instanceof PublicKeyRingBlob)
             case ASCII_ARMOR:
                 result.stream = new ArmoredInputStream(result.stream); // fall thru
             case PGP:
-                return new BcPGPObjectFactory(result.stream).iterator();
+                return new PGPObjectFactory(result.stream, JCAContextHelper.getJcaKeyFingerprintCalculator()).iterator();
             case KEYBOX:
-                return new BcKeyBox(result.stream).getKeyBlobs().iterator();
+                try {
+                    return JCAContextHelper.getJcaKeyBoxBuilder().build(result.stream).getKeyBlobs().iterator();
+                } catch (NoSuchProviderException | NoSuchAlgorithmException e) {
+                    throw new RuntimeException(e);
```

Comment:
Instead of wrapping this with a `RuntimeException`, wrap this with a `PGPException` (so users can catch a `PGPException` in their own code if that want to handle anything PGP-related going wrong when they use JPGPJ).